.github/workflows/tests.yml

```diff
@@ -76,3 +76,25 @@ jobs:
         # XDG_RUNTIME_DIR is set.
         # TODO: figure out what exactly podman needs
         sudo -E XDG_RUNTIME_DIR= pytest-3 -s -vv
+
+  integration-macos:
+    name: "Integration macos"
+    # needed to get latest cpu
+    runs-on: macos-13
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Is the checkout action uses this ref by default? Is that addition necessary?

Reply:
Thank you, that is a nice catch. TBH I just cargo-culted this from the rest of the file. I will clean it up everywhere :)

Reply:
Actually, I looked at this more closely and there is a subtle difference. The default behavior is AFAICT that a merge commit of main and the PR is generated and checked out. With our current setup we only get the PR HEAD [1].

I have no strong opinion either way but would prefer to consult with the team first (as I'm relatively new and lack a lot of the history) what the rational for this decision is and if it needs changing I will do it in a separate PR. Hope that is okay with you :)

[1] https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit

Reply:
I think that one advantage is of using a PR head is that you clearly know what commit is being tested. But it's matter of taste, tbh.

I'm more interested if this works correctly when run on a merge queue, it looks a bit fishy to me tbh.

Reply:
That's a good point, because on the merge queue, the `github.event.pull_request` should be empty, so I guess `ref` is unset then.
Comparing https://github.com/osbuild/bootc-image-builder/actions/runs/7384902395/job/20088639130#step:2:2 (checkout action for shellcheck on a PR) with https://github.com/osbuild/bootc-image-builder/actions/runs/7386384027/job/20092854868#step:2:2 (checkout action for shellcheck on the merge queue) seems to support this.

Reply:
If I read the room correctly here it sounds like we do want to move to testing the merge commit instead (at least on systems that use a merge queue)? Fwiw, I tend to agree but have no strong opinion either way. I do care that we do it consistently though :) Unless someone here disagrees I will prepare PRs to make this change consistently.

Reply:
I will do this in a followup if that's okay? Would like to not block on it for this PR.